drivers/sensor/icm42605/icm42605.c

```diff
@@ -0,0 +1,463 @@
+			} else {
+				drv_data->gyro_sf = val->val1;
+			}
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+		break;
+	default:
+		LOG_ERR("Not support");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int icm42605_attr_get(const struct device *dev,
+			     enum sensor_channel chan,
+			     enum sensor_attribute attr,
+			     struct sensor_value *val)
+{
+	const struct icm42605_data *drv_data = dev->data;
+
+	__ASSERT_NO_MSG(val != NULL);
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			val->val1 = drv_data->accel_hz;
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			val->val1 = drv_data->accel_sf;
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+
+		break;
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			val->val1 = drv_data->gyro_hz;
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			val->val1 = drv_data->gyro_sf;
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+
+		break;
+
+	default:
+		LOG_ERR("Not support");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int icm42605_data_init(struct icm42605_data *data,
+			      const struct icm42605_config *cfg)
+{
+	data->accel_x = 0;
+	data->accel_y = 0;
+	data->accel_z = 0;
+	data->temp = 0;
+	data->gyro_x = 0;
+	data->gyro_y = 0;
+	data->gyro_z = 0;
+	data->accel_hz = cfg->accel_hz;
+	data->gyro_hz = cfg->gyro_hz;
+
+	data->accel_sf = cfg->accel_fs;
+	data->gyro_sf = cfg->gyro_fs;
+
+	data->tap_en = false;
+	data->sensor_started = false;
+
+	return 0;
+}
+
+
+static int icm42605_init(const struct device *dev)
+{
+	struct icm42605_data *drv_data = dev->data;
+	const struct icm42605_config *cfg = dev->config;
+
+	drv_data->spi = device_get_binding(cfg->spi_label);
+	if (!drv_data->spi) {
+		LOG_ERR("SPI device not exist");
+		return -ENODEV;
+	}
+
+	drv_data->spi_cs.gpio_dev = device_get_binding(cfg->gpio_label);
+	if (!drv_data->spi_cs.gpio_dev) {
+		LOG_ERR("GPIO device not exist");
+		return -ENODEV;
+	}
+
+	drv_data->spi_cs.gpio_dev = device_get_binding(cfg->gpio_label);
+	drv_data->spi_cs.gpio_pin = cfg->gpio_pin;
+	drv_data->spi_cs.gpio_dt_flags = cfg->gpio_dt_flags;
+	drv_data->spi_cs.delay = 0U;
+
+	drv_data->spi_cfg.frequency = cfg->frequency;
+	drv_data->spi_cfg.slave = cfg->slave;
+	drv_data->spi_cfg.operation = (SPI_OP_MODE_MASTER | SPI_MODE_CPOL |
+			SPI_MODE_CPHA | SPI_WORD_SET(8) | SPI_LINES_SINGLE |
+			SPI_TRANSFER_MSB);
+	drv_data->spi_cfg.cs = &drv_data->spi_cs;
+
+	icm42605_spi_init(drv_data->spi, &drv_data->spi_cfg);
+	icm42605_data_init(drv_data, cfg);
+	icm42605_sensor_init(dev);
+
+	drv_data->accel_sensitivity_shift = 14 - 3;
+	drv_data->gyro_sensitivity_x10 = icm42605_gyro_sensitivity_x10[3];
+
+	drv_data->icm42605_tap_fetch = icm42605_tap_fetch;
```

Comment:
Why do you need this? It looks like you could just call the function directly in line 90

Reply:
icm42605_tap_fetch() is called in icm42605_trigger.c file. so I set function pointer of icm42605_tap_fetch()

Reply:
Why can't you call it directly there too?

Reply:
@MaureenHelm Fixed